pyoomph/c_output.py

```diff
@@ -185,17 +186,17 @@ def emit_instruction(self, ins: ir.Instruction) -> str:
             """
 
         if isinstance(ins, ir.IsNull):
-            return f"{self.emit_var(ins.result)} = {self.emit_var(ins.var)}.isnull;\n"
+            return f"{self.emit_var(ins.result)} = IS_NULL({self.emit_var(ins.var)});\n"
 
         if isinstance(ins, ir.SetToNull):
             if isinstance(ins.var.type, UnionType):
+                if (
+                    ins.var.type.generic_origin is not None
+                    and ins.var.type.generic_origin.generic is OPTIONAL
+                ):
+                    c_type = self.file_pair.emit_type(ins.var.type)
+                    return f"{self.emit_var(ins.var)} = ({c_type}){{0}};\n"
```

Comment:
This should be cleaned up by e.g. replacing `SetToNull` with two new IR operations, one for unions and one for what `SetToNull` usually does